FogCreek/Glitch-Community
# harmless-path: Update homepage + pupdates to handle rich text content and update /{page}/preview to show drafts from new CMS

## Links
* [harmless-path.glitch.me](harmless-path.glitch.me)
* [Spec](https://docs.google.com/document/d/1t0t2vcrqKU9M7MYR9mmACYwqfK_WFCifwpuyTw5gcRQ/edit#heading=h.2aws1x61wxk4)

*I'll be writing more technical documentation about how the new CMS works at [cms.glitch.me](https://cms.glitch.me), but these changes are minor, and should be reviewable without those docs*

## GIF/Screenshots:
**Previewing draft versions of the homepage**
<img width="400" src="https://user-images.githubusercontent.com/7584833/66598740-b27dd900-eb6f-11e9-9d16-b3d73e720b2a.gif" />

**Previewing draft versions of pupdates**
<img width="300" src="https://user-images.githubusercontent.com/7584833/66598750-bb6eaa80-eb6f-11e9-959f-036c54d89698.png" />

## Changes:
* ran `npx sort-package-json`
* Refactor the `/api/pupdate` and `/api/home` endpoints to be one endpoint
* Remove "Publish" buttons from`<PreviewContainer>`. Publishing now happens in our new CMS, Prismic
* Update `<DataLoader>` to reload when the `get` prop changes, which enables...
* Add a `<select>` dropdown to switch between draft versions on /index/preview and /pupdates/preview
* Update homepage and pupdates renderers to handle rich text from our CMS
   * See the the changes to `src/curated/home.json`, you'll notice that it's HTML strings instead of raw markdown
  * This required using `dangerouslySetInnerHTML` and disabling that eslint rule on each line we use it

## How To Test:
It's a bit tricky for y'all to test this, but I'm happy to step through my QA checklist with folks in-person or on video chat!

- Create a draft release and make some changes. It is viewable from [/index/preview](https://harmless-path.glitch.me/index/preview) and [/pupdates/preview](https://harmless-path.glitch.me/pupdates/preview)
- Delete a draft release. It should be deleted from cms.glitch.me and no longer viewable on [/index/preview](https://harmless-path.glitch.me/index/preview) and [/pupdates/preview](https://harmless-path.glitch.me/pupdates/preview)
- Publish changes to the homepage. They should show up at [harmless-path.glitch.me](harmless-path.glitch.me)
- Publish a new pupdate, it should show up in the log
- Delete ^this pupdate, it should be gone

## Feedback I'm looking for:
- Do we still feel ok about using `dangerouslySetInnerHTML`?
- Are there any unintended consequences of adding `get` to [this dependency array](https://github.com/FogCreek/Glitch-Community/compare/harmless-path?expand=1#diff-277bb733defee34992986736b0ab8760R32)?

## Things left to do before deploying:
- [x] Set up accounts for marketing team on Prismic
- [ ] Update cms.glitch.me to point to glitch.com instead of harmless-path.glitch.me
- [x] Confirm that all the live data in Prismic is up-to-date with glitch.com
- [x] Finish documentation in cms.glitch.me, announce to company
